pkg/client/client.go

```diff
@@ -185,9 +205,15 @@ func newClient(config *rest.Config, options Options) (*client, error) {
 		return c, nil
 	}
 
+	// Default to looking up items when missing in the cache
+	if options.Cache.MissPolicy == "" {
+		options.Cache.MissPolicy = CacheMissPolicyLiveLookup
```

Comment:
I thought we wanted to default to fail?

Reply:
I considered this but it seems like the more surprising / breaking change. Brings it more inline with k8s lister / live client split, though - is that why you think fail should be default?

Reply:
I think falling back might lead to questions, because based on if there is a cache or not the request will get a fast or slow response that is incoherent or not - That is less likely to happen if we only do that when ppl explicitly opted into this

Reply:
Sec, what's the current behavior?

Reply:
There is no possibility of this happening currently, as the cache always backfills.

Reply:
Perhaps we should re-name this from "miss" policy in both cases, since it's not truly a cache miss in the 404 sense, but a "cache not configured to hold this resource type" miss.

Reply:
Refactored to make this wording more clear.